src/extensions/default/PrefsCodeHints/main.js

```diff
@@ -0,0 +1,385 @@
+        ExtensionUtils      = brackets.getModule("utils/ExtensionUtils"),
+        EditorManager       = brackets.getModule("editor/EditorManager"),
+        LanguageManager     = brackets.getModule("language/LanguageManager"),
+        JSONUtils           = brackets.getModule("language/JSONUtils"),
+        Strings             = brackets.getModule("strings"),
+        data                = JSON.parse(require("text!data.json")),
+        isPrefDocument      = false,
+        isPrefHintsEnabled  = false;
+    
+    var stringMatcherOptions = {
+        preferPrefixMatches: true
+    };
+    
+    // List of parent keys for which no key hints will be provided.
+    var parentKeyBlacklist = [
+        "language.fileExtensions",
+        "language.fileNames",
+        "path"
+    ];
+    
+    // Define a preference for code hinting.
+    PreferencesManager.definePreference("codehint.PrefHints", "boolean", true);
+    
+    /**
+     * @private
+     *
+     * Determines whether or not the current document is a preferences document and
+     * user has enabled code hints
+     * 
+     * @return {Boolean}
+     */
+    function _isPrefHintsEnabled() {
+        return (isPrefDocument &&
+                PreferencesManager.get("showCodeHints") !== false &&
+                PreferencesManager.get("codehint.PrefHints") !== false);
+    }
+    
+    /**
+     * Determines whether or not the name of a file matches the preferences files
+     *
+     * @param {!Document} document
+     * @return {Boolean}
+     */
+    function _isPrefDocument(document) {
+        return (/^\.?brackets\.json$/).test(document.file._name);
+    }
+    
+    // Set listeners on preference and editor changes.
+    PreferencesManager.on("change", "showCodeHints", function () {
+        isPrefHintsEnabled = _isPrefHintsEnabled();
+    });
+    PreferencesManager.on("change", "codehint.PrefHints", function () {
+        isPrefHintsEnabled = _isPrefHintsEnabled();
+    });
+    EditorManager.on("activeEditorChange", function (e, editor) {
+        if (editor) {
+            isPrefDocument = _isPrefDocument(editor.document);
+        }
+        isPrefHintsEnabled = _isPrefHintsEnabled();
+    });
+    
+    /*
+     * Returns a sorted and formatted list of hints with the query substring
+     * highlighted.
+     * 
+     * @param {Array.<Object>} hints - the list of hints to format
+     * @param {string} query - querystring used for highlighting matched
+     *      portions of each hint
+     * @return {Array.jQuery} sorted Array of jQuery DOM elements to insert
+     */
+    function formatHints(hints, query) {
+        
+        var hasMetadata = hints.some(function (token) {
+            return token.type || token.description;
+        });
+        
+        StringMatch.basicMatchSort(hints);
+        return hints.map(function (token) {
+            var $hintItem = $("<span>").addClass("brackets-pref-hints"),
```

Comment:
![codehints](https://cloud.githubusercontent.com/assets/5336369/7861095/b8e9bc26-056b-11e5-9dff-53e2fb202cb5.gif)
The width of the selection keeps jumping; Would it be possible to fix the width and wrap the hint description. Maybe with <divs> and some css with `display: block;    visibility: visible;    white-space: pre-wrap;`